### PR TITLE
Correctly invalidate I-cache on aarch64 SMP [SELFOUR-2830]

### PR DESCRIPTION
**Pull request by bennoleslie**
See: [SELFOUR-2830] for details.

On SMP configuration the `ic ialluis` instruction must be used
to ensure the I-cache on all cores is invalidated.

Signed-off-by: Ben Leslie <benno@brkawy.com>

[SELFOUR-2830]: https://sel4.atlassian.net/browse/SELFOUR-2830

### CODE REVIEW

File: include/arch/arm/arch/64/mode/machine.h [hunk 1]
```diff
@@ -304,7 +304,11 @@ static inline void invalidateByVA_I(vptr_t vaddr, paddr_t paddr)
 
 static inline void invalidate_I_PoU(void)
 {
+#if CONFIG_MAX_NUM_NODES > 1
+    asm volatile("ic ialluis");
+#else
     asm volatile("ic iallu");
```
[the remark this answers is not in the document]
Comment (bennoleslie) [reply include/arch/arm/arch/64/mode/machine.h:310]: I think such an approach is desirable, but more complex to implement and validate.

For now this change ensures correctness (while the behavior is currently incorrect), and there can be additional optimizations in the future.

While it is use case specific, I don't expect this to be a common path which is hit, so the performance overhead isn't too bad.

If that's the only objection I feel this should be merged and additional PRs made for optimizing.

[the remark this answers is not in the document]
Comment (yyshen) [reply include/arch/arm/arch/64/mode/machine.h:310]: another issue is if icache is invalidated IS, an ISB should be executed on the cores to ensure refetching, while the ISB does not have broadcast semantic ...

[the remark this answers is not in the document]
Comment (kent-mcleod) [reply include/arch/arm/arch/64/mode/machine.h:310]: As the kernel makes all mappings inner shareable the other cache operations by VA are already doing this.  So this change is just making it consistent I believe.

**Review by kent-mcleod — APPROVED** [review]

File: include/arch/arm/arch/64/mode/machine.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (lsf37) [reply include/arch/arm/arch/64/mode/machine.h:310]: The experiments in [SELFOUR-2830] seem to confirm that this is the right thing to do, but I have to rely on the experts for this one. @yyshen are you happy with Kent's argument?

**Comment by lsf37** [comment]
@ssrg-bamboo `test`

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

**Comment by ssrg-bamboo** [comment]
## Test results for commit 563e063a88b7e82f6cea255d7ca64895ecbbf1e3

### Summary

| Test name | Result |
| --- | --- |
| Code Quality - C Parser - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | FAILED |
| Code Quality - Check licenses - seL4 - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | SUCCESSFUL |
| Code Quality - Reuse License Check - seL4 - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | SUCCESSFUL |
| Code Quality - Signed-off-by - seL4 - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | SUCCESSFUL |
| Code Quality - Style - seL4 - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | SUCCESSFUL |
| Dev seL4 tests - seL4 PRs: Kernel Compile - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | SUCCESSFUL |
| Dev seL4 tests - seL4 PRs: Preprocess - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | SUCCESSFUL |
| Dev seL4 tests - seL4 PRs: compile seL4test - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | FAILED |

### Detailed failure logs

| Project | Stage | Result summary |
| --- | --- | --- |
| Code Quality - C Parser - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | Set GitHub Success/Fail [Run C-Parser] | 1 of 1 job failed |

**Job: Set notifications [Run C-Parser]**
Build logs

    22-Apr-2021 14:08:44	Build Code Quality - C Parser - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 - Set notifications [Run C-Parser] #1 (CQ-CPARSE456-RUNCPARSERGHSTATUS-1) started building on agent kzm, bamboo version: 7.1.4
    22-Apr-2021 14:08:44	
    22-Apr-2021 14:08:44	Build working directory is /var/atlassian/application-data/bamboo/xml-data/build-dir/6258689/CQ-CPARSE456-RUNCPARSERGHSTATUS
    22-Apr-2021 14:08:44	Executing build Code Quality - C Parser - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 - Set notifications [Run C-Parser] #1 (CQ-CPARSE456-RUNCPARSERGHSTATUS-1)
    22-Apr-2021 14:08:44	Preparing artifact 'name.properties' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/6258689/CQ-CPARSE456-RUNCPARSERGHSTATUS/. (location: .)
    22-Apr-2021 14:08:44	Artifact 'name.properties' prepared in 13.47 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/6258689/CQ-CPARSE456-RUNCPARSERGHSTATUS/./name.properties
    22-Apr-2021 14:08:44	Preparing artifact 'sha.properties' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/6258689/CQ-CPARSE456-RUNCPARSERGHSTATUS/. (location: .)
    22-Apr-2021 14:08:44	Artifact 'sha.properties' prepared in 7.338 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/6258689/CQ-CPARSE456-RUNCPARSERGHSTATUS/./sha.properties
    22-Apr-2021 14:08:44	Preparing artifact 'success_Run_C-Parser_AARCH32CPARSE' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/6258689/CQ-CPARSE456-RUNCPARSERGHSTATUS/. (location: .)
    22-Apr-2021 14:08:44	Failure in artifact preparation phase during processing of: Subscription for Non required shared artifact: [success_Run_C-Parser_AARCH32CPARSE], pattern: [*.Run_C-Parser.success] anchored at: [.], destination: [.]
    22-Apr-2021 14:08:44	1 error(s) found when performing pre-build actions. 
    22-Apr-2021 14:08:44	Running on server: post build plugin 'NCover Results Collector'
    22-Apr-2021 14:08:44	Running on server: post build plugin 'Build Hanging Detection Configuration'
    22-Apr-2021 14:08:44	Running on server: post build plugin 'Clover Delta Calculator'
    22-Apr-2021 14:08:44	Running on server: post build plugin 'Maven Dependencies Postprocessor'
    22-Apr-2021 14:08:44	All post build plugins have finished
    22-Apr-2021 14:08:44	Generating build results summary...
    22-Apr-2021 14:08:44	Saving build results to disk...
    22-Apr-2021 14:08:44	Store variable context...
    22-Apr-2021 14:08:44	Indexing build results...
    22-Apr-2021 14:08:44	Finished building CQ-CPARSE456-RUNCPARSERGHSTATUS-1.


| Project | Stage | Result summary |
| --- | --- | --- |
| Dev seL4 tests - seL4 PRs: compile seL4test - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | Set GitHub Success/Fail [Build seL4test] | 1 of 1 job failed |

**Job: Set notifications [Build seL4test]**
Build logs

    22-Apr-2021 14:15:43	Build Dev seL4 tests - seL4 PRs: compile seL4test - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 - Set notifications [Build seL4test] #1 (SEL4-PHT1616-BUILDSEL4TESTGHSTATUS-1) started building on agent rockpro64a, bamboo version: 7.1.4
    22-Apr-2021 14:15:43	
    22-Apr-2021 14:15:43	Build working directory is /var/atlassian/application-data/bamboo/xml-data/build-dir/237076577/SEL4-PHT1616-BUILDSEL4TESTGHSTATUS
    22-Apr-2021 14:15:43	Executing build Dev seL4 tests - seL4 PRs: compile seL4test - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 - Set notifications [Build seL4test] #1 (SEL4-PHT1616-BUILDSEL4TESTGHSTATUS-1)
    22-Apr-2021 14:15:43	Preparing artifact 'name.properties' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/237076577/SEL4-PHT1616-BUILDSEL4TESTGHSTATUS/. (location: .)
    22-Apr-2021 14:15:43	Artifact 'name.properties' prepared in 11.44 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/237076577/SEL4-PHT1616-BUILDSEL4TESTGHSTATUS/./name.properties
    22-Apr-2021 14:15:43	Preparing artifact 'sha.properties' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/237076577/SEL4-PHT1616-BUILDSEL4TESTGHSTATUS/. (location: .)
    22-Apr-2021 14:15:43	Artifact 'sha.properties' prepared in 9.027 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/237076577/SEL4-PHT1616-BUILDSEL4TESTGHSTATUS/./sha.properties
    22-Apr-2021 14:15:43	Preparing artifact 'success_Build_seL4test_CLANGARMV6AHWBUILDID' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/237076577/SEL4-PHT1616-BUILDSEL4TESTGHSTATUS/. (location: .)
    22-Apr-2021 14:15:43	Artifact 'success_Build_seL4test_CLANGARMV6AHWBUILDID' prepared in 7.820 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/237076577/SEL4-PHT1616-BUILDSEL4TESTGHSTATUS/./CLANGARMV6AHWBUILDID.Build_seL4test.success
    22-Apr-2021 14:15:43	Preparing artifact 'success_Build_seL4test_CLANGARMV7AHWBUILD1ID' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/237076577/SEL4-PHT1616-BUILDSEL4TESTGHSTATUS/. (location: .)
    22-Apr-2021 14:15:43	Failure in artifact preparation phase during processing of: Subscription for Non required shared artifact: [success_Build_seL4test_CLANGARMV7AHWBUILD1ID], pattern: [*.Build_seL4test.success] anchored at: [.], destination: [.]
    22-Apr-2021 14:15:43	1 error(s) found when performing pre-build actions. 
    22-Apr-2021 14:15:43	Running on server: post build plugin 'NCover Results Collector'
    22-Apr-2021 14:15:43	Running on server: post build plugin 'Build Hanging Detection Configuration'
    22-Apr-2021 14:15:43	Running on server: post build plugin 'Clover Delta Calculator'
    22-Apr-2021 14:15:43	Running on server: post build plugin 'Maven Dependencies Postprocessor'
    22-Apr-2021 14:15:43	All post build plugins have finished
    22-Apr-2021 14:15:43	Generating build results summary...
    22-Apr-2021 14:15:43	Saving build results to disk...
    22-Apr-2021 14:15:43	Store variable context...
    22-Apr-2021 14:15:43	Indexing build results...
    22-Apr-2021 14:15:43	Finished building SEL4-PHT1616-BUILDSEL4TESTGHSTATUS-1.


This is the most I can report on right now, sorry!

File: include/arch/arm/arch/64/mode/machine.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (yyshen) [reply include/arch/arm/arch/64/mode/machine.h:310]: the memory region is mapped as inner shareable, so if the SMP bit is enabled by the cores, data cache maintenances are broadcasted to all cores and DSB ensures the operations are done by all cores. As I mentioned above, instruction cache maintenances need ISB to refetch instructions, but ISB is not broadcasted. So, other cores could still run the old instructions, but this does not always cause problems. 
Copied from Arm ARM (let me know if this is not allowed).
image

Using the broadcast version is better, but I think an IPI version should be implemented later.

**Review by yyshen — APPROVED** [review]

**Comment by lsf37** [comment]
I hope a rerun one gives more information. Otherwise this is good to merge.

@ssrg-bamboo `test`

**Review by lsf37 — APPROVED** [review]

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

**Comment by bennoleslie** [comment]
@ssrg-bamboo test

does this work?

**Comment by bennoleslie** [comment]
Kick of a re-test

@ssrg-bamboo `test`

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

**Comment by ssrg-bamboo** [comment]
## Test results for commit 945c5fc539acc4cd082a5bcc29eb893598bbef02

### Summary

#### NOTE: some test results have been updated!

| Test name | Result |
| --- | --- |
| Code Quality - C Parser - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | SUCCESSFUL |
| Code Quality - Check licenses - seL4 - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | SUCCESSFUL |
| Code Quality - Reuse License Check - seL4 - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | SUCCESSFUL |
| Code Quality - Signed-off-by - seL4 - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | FAILED |
| Code Quality - Style - seL4 - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | SUCCESSFUL |
| Dev seL4 tests - seL4 PRs: Kernel Compile - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | SUCCESSFUL |
| Dev seL4 tests - seL4 PRs: Preprocess - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | SUCCESSFUL |
| Dev seL4 tests - seL4 PRs: compile seL4test - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | SUCCESSFUL |

### Detailed failure logs

| Project | Stage | Result summary |
| --- | --- | --- |
| Code Quality - Signed-off-by - seL4 - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 | Check all commits have been Signed-Off-by | 1 of 1 job failed |

**Job: signedoff**
Build logs

    28-Apr-2021 15:38:18	Build Code Quality - Signed-off-by - seL4 - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 - signedoff #2 (CQ-SEL4SIGNEDOFF210-SEL4SIGNEDOFFSIGNEDOFF-2) started building on agent A local agent 1, bamboo version: 7.1.4
    28-Apr-2021 15:38:18	I run on the Bamboo server!
    28-Apr-2021 15:38:18	Build working directory is /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4SIGNEDOFF210-SEL4SIGNEDOFFSIGNEDOFF
    28-Apr-2021 15:38:18	Executing build Code Quality - Signed-off-by - seL4 - EXT-BreakawayConsulting-invalidate-aarch64-smp-351 - signedoff #2 (CQ-SEL4SIGNEDOFF210-SEL4SIGNEDOFFSIGNEDOFF-2)
    28-Apr-2021 15:38:18	Preparing artifact 'code' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4SIGNEDOFF210-SEL4SIGNEDOFFSIGNEDOFF/. (location: .)
    28-Apr-2021 15:38:18	Artifact 'code' prepared in 10.44 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4SIGNEDOFF210-SEL4SIGNEDOFFSIGNEDOFF/./code.tar.gz
    28-Apr-2021 15:38:18	Running pre-build action: VCS Version Collector
    28-Apr-2021 15:38:18	Starting task 'Untar code' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    28-Apr-2021 15:38:18	+ tar xf code.tar.gz --recursive-unlink
    28-Apr-2021 15:38:18	Finished task 'Untar code' with result: Success
    28-Apr-2021 15:38:18	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    28-Apr-2021 15:38:18	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4SIGNEDOFF210-SEL4SIGNEDOFFSIGNEDOFF
    28-Apr-2021 15:38:18	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    28-Apr-2021 15:38:18	Starting task 'Check commits have been signed-off' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    28-Apr-2021 15:38:18	+ cd seL4
    28-Apr-2021 15:38:18	+++ git merge-base 945c5fc539acc4cd082a5bcc29eb893598bbef02 d486a16961882fc86c19bab149f016bae2f86e42
    28-Apr-2021 15:38:18	++ git rev-list --invert-grep --grep=Signed-off-by: d486a16961882fc86c19bab149f016bae2f86e42..945c5fc539acc4cd082a5bcc29eb893598bbef02
    28-Apr-2021 15:38:18	+ commits_not_signed_off=945c5fc539acc4cd082a5bcc29eb893598bbef02
    28-Apr-2021 15:38:18	+ '[' '!' -z 945c5fc539acc4cd082a5bcc29eb893598bbef02 ']'
    28-Apr-2021 15:38:18	+ echo 'The follow commits need to be signed off:'
    28-Apr-2021 15:38:18	The follow commits need to be signed off:
    28-Apr-2021 15:38:18	+ echo 945c5fc539acc4cd082a5bcc29eb893598bbef02
    28-Apr-2021 15:38:18	945c5fc539acc4cd082a5bcc29eb893598bbef02
    28-Apr-2021 15:38:18	+ exit 1
    28-Apr-2021 15:38:18	Failing task since return code of [/var/atlassian/application-data/bamboo/temp/CQ-SEL4SIGNEDOFF210-SEL4SIGNEDOFFSIGNEDOFF-2-ScriptBuildTask-9125772686672857329.sh] was 1 while expected 0
    28-Apr-2021 15:38:18	Finished task 'Check commits have been signed-off' with result: Failed
    28-Apr-2021 15:38:18	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    28-Apr-2021 15:38:18	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4SIGNEDOFF210-SEL4SIGNEDOFFSIGNEDOFF
    28-Apr-2021 15:38:18	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    28-Apr-2021 15:38:18	Starting task 'Run dco code insights' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    28-Apr-2021 15:38:18	+ cd seL4
    28-Apr-2021 15:38:18	++ echo ssh://git@bitbucket.ts.data61.csiro.au:7999/sel4/sel4.git
    28-Apr-2021 15:38:18	++ cut -d/ -f4
    28-Apr-2021 15:38:18	+ bb_proj=sel4
    28-Apr-2021 15:38:18	+ /scripts/bb_code_insights/create_code_insights.sh git-commits-signed-off sel4 seL4
    28-Apr-2021 15:38:18	++ dirname /scripts/bb_code_insights/create_code_insights.sh
    28-Apr-2021 15:38:18	+ SCRIPT_DIR=/scripts/bb_code_insights
    28-Apr-2021 15:38:18	+ BBS_URL=https://bitbucket.ts.data61.csiro.au
    28-Apr-2021 15:38:18	+ BBS_PROJECT=sel4
    28-Apr-2021 15:38:18	+ BBS_REPO=seL4
    28-Apr-2021 15:38:18	+ REPORT_TO_RUN=git-commits-signed-off
    28-Apr-2021 15:38:18	++ git rev-parse HEAD
    28-Apr-2021 15:38:18	+ COMMIT_ID=945c5fc539acc4cd082a5bcc29eb893598bbef02
    28-Apr-2021 15:38:18	+ REPORT_KEY=git-commits-signed-off.report
    28-Apr-2021 15:38:18	+ set +x
    28-Apr-2021 15:38:18	+ report_file=git-commits-signed-off_report.json
    28-Apr-2021 15:38:18	+ annotations_file=git-commits-signed-off_annotations.json
    28-Apr-2021 15:38:18	+ rm -f git-commits-signed-off_report.json git-commits-signed-off_annotations.json
    28-Apr-2021 15:38:18	+ /scripts/bb_code_insights/git-commits-signed-off
    28-Apr-2021 15:38:18	Running: git merge-base 945c5fc539acc4cd082a5bcc29eb893598bbef02 d486a16961882fc86c19bab149f016bae2f86e42
    28-Apr-2021 15:38:18	d486a16961882fc86c19bab149f016bae2f86e42
    28-Apr-2021 15:38:18	Running: git rev-list --invert-grep --grep=Signed-off-by: d486a16961882fc86c19bab149f016bae2f86e42..945c5fc539acc4cd082a5bcc29eb893598bbef02
    28-Apr-2021 15:38:18	945c5fc539acc4cd082a5bcc29eb893598bbef02
    28-Apr-2021 15:38:18	Identified bad commits:
    28-Apr-2021 15:38:18	[]
    28-Apr-2021 15:38:18	Reporting the results to BitBucket, but doing so *secretly*, so people don't see the access token
    28-Apr-2021 15:38:18	+ echo 'Reporting the results to BitBucket, but doing so *secretly*, so people don'\''t see the access token'
    28-Apr-2021 15:38:18	+ set +x
    28-Apr-2021 15:38:18	+ '[' -f git-commits-signed-off_report.json ']'
    28-Apr-2021 15:38:18	+ cat git-commits-signed-off_report.json
    28-Apr-2021 15:38:18	+ '[' -f git-commits-signed-off_annotations.json ']'
    28-Apr-2021 15:38:18	{"data":[{"title":"Total number of commits not signed-off","value":0,"type":"NUMBER"}],"createdDate":1619588298918,"details":"All good B-)","key":"git-commits-signed-off.report","result":"PASS","title":"Signed-off commits"}{"title": "Signed-off commits", "data": [{"title": "Total number of commits not signed-off", "value": 0, "type": "NUMBER"}], "details": "All good B-)", "result": "PASS"}
    28-Apr-2021 15:38:18	Finished task 'Run dco code insights' with result: Success
    28-Apr-2021 15:38:18	Running post build plugin 'NCover Results Collector'
    28-Apr-2021 15:38:18	Running post build plugin 'Artifact Copier'
    28-Apr-2021 15:38:18	Running post build plugin 'npm Cache Cleanup'
    28-Apr-2021 15:38:18	Running post build plugin 'Clover Results Collector'
    28-Apr-2021 15:38:18	Running post build plugin 'Docker Container Cleanup'
    28-Apr-2021 15:38:18	Successfully removed working directory at '/var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4SIGNEDOFF210-SEL4SIGNEDOFFSIGNEDOFF'
    28-Apr-2021 15:38:18	Finalising the build...
    28-Apr-2021 15:38:18	Stopping timer.
    28-Apr-2021 15:38:18	Build CQ-SEL4SIGNEDOFF210-SEL4SIGNEDOFFSIGNEDOFF-2 completed.
    28-Apr-2021 15:38:18	Running on server: post build plugin 'NCover Results Collector'
    28-Apr-2021 15:38:18	Running on server: post build plugin 'Build Hanging Detection Configuration'
    28-Apr-2021 15:38:18	Running on server: post build plugin 'Clover Delta Calculator'
    28-Apr-2021 15:38:18	Running on server: post build plugin 'Maven Dependencies Postprocessor'
    28-Apr-2021 15:38:18	All post build plugins have finished
    28-Apr-2021 15:38:18	Generating build results summary...
    28-Apr-2021 15:38:18	Saving build results to disk...
    28-Apr-2021 15:38:19	Store variable context...
    28-Apr-2021 15:38:19	Indexing build results...
    28-Apr-2021 15:38:19	Finished building CQ-SEL4SIGNEDOFF210-SEL4SIGNEDOFFSIGNEDOFF-2.


This is the most I can report on right now, sorry!

**Comment by lsf37** [comment]
(the sign-off check on bamboo fails on the temporary update-merge commits; all good)

**Pull request merged**